jujugui/static/gui/src/app/app.js

```diff
@@ -1894,7 +1894,12 @@ YUI.add('juju-gui', function(Y) {
       @param {String} password The password for the new environment.
     */
     switchEnv: function(socketUrl, username, password) {
```

Comment:
You can actually do something cool here for the reconnect while also allowing it to be a param
```JavaScript
  switchEnv: function(socketUrl, username, password, reconnect=!!socketUrl) {
```